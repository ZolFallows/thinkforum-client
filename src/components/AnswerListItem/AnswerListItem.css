.Answer_box {
    margin-top: 5px;
    border: 1px solid #ececec;
}

.Answer_header {
    padding: 0 5px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    border-bottom: 1px solid #ececec;
    background: #ececec;
    /* font-size: 1rem;
    font-weight: 500;
    text-decoration: none;  */
}

.Answer_header .full_name {
    margin: 0;
    font-size: 1rem;
    font-weight: 500;
}

.Answer_box .text {
    padding: 5px;
}