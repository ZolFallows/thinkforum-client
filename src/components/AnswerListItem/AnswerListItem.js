import React, { Component } from 'react'
import Moment from 'react-moment'
import parse from 'html-react-parser'
import Showdown from "showdown"

export default class AnswerListItem extends Component {
    constructor(props) {
        super(props)
        this.converter = new Showdown.Converter({
          tables: true,
          simplifiedAutoLink: true
        })
      }

    render(){
        const { answer } = this.props
        return (
            <div className="Answer_box">
                <div className="Answer_header">
                    <p>{answer.user.full_name}</p>
                    <p><Moment fromNow>{answer.date_created}</Moment></p>
                </div>
                <div className="text">
                    {parse(this.converter.makeHtml(answer.text))}
                </div>
            </div>
        )
    }
}