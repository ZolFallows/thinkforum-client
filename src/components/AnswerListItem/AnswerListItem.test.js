
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AnswerListItem from './AnswerListItem'

const sectionProp = {
                        "id": 8,
                        "text": "Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.",
                        "user": {
                            "id": 3,
                            "user_name": "c.bloggs",
                            "full_name": "Charlie Bloggs",
                            "date_created": "2019-03-27T20:27:19.481Z"
                        },
                        "date_created": "2019-03-27T20:27:19.481Z"
                    }


describe(`AnswerListItem Component`, () => {
    it('Renders without crashing', () => {
        shallow(<AnswerListItem/>)
      });

    it('renders empty given no AnswerListItem without errors', () => {
        const wrapper = shallow(<AnswerListItem />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders given AnswerListItem without errors', () => {
        const wrapper = shallow(<AnswerListItem answer={sectionProp} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})
