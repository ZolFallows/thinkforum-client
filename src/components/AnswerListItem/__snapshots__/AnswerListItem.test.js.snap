// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerListItem Component renders empty given no AnswerListItem without errors 1`] = `<Fragment />`;

exports[`AnswerListItem Component renders given AnswerListItem without errors 1`] = `
<Fragment>
  <div
    className="Answer_box"
  >
    <div
      className="Answer_header"
    >
      <p
        className="full_name"
      >
        Charlie Bloggs
      </p>
      <p
        className="date_created"
      >
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element="time"
          filter={[Function]}
          fromNow={true}
          interval={60000}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          2019-03-27T20:27:19.481Z
        </t>
      </p>
    </div>
    <div
      className="text"
    >
      <p>
        Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.
      </p>
    </div>
  </div>
</Fragment>
`;
